joinview 생성

leten02 opened the pull request:
회원가입 기본 정보 입력 페이지 구현(1차) 

구현된 목록
- 초대코드 입력 및 QR스캔
- 학교 선택
- 기수 선택
- 이름 입력
- 닉네임 입력
- 파트 선택
- 개인 정보 동의

leten02 committed:
joinview 생성

SeungWon1125 reviewed (commented):
확인했습니다!

yaeun commented:
확인했습니다!

yaeun commented on UMC-iOS/UMC-iOS/Presentation/Login/Helper/JoinNavigationButton.swift:
```diff
@@ -0,0 +1,38 @@
+//
+
+import SwiftUI
+
```
//확인했습니다

kyxxgsoo commented on UMC-iOS/UMC-iOS/Presentation/Login/JoinView/SelectGeneration.swift:
```diff
@@ -0,0 +1,74 @@
+                    }
+                    
+
+                    HStack(spacing: 8){
```
VStack과 HStack을 적절히 사용해서 중복된 코드 제거해주세요

kyxxgsoo commented on UMC-iOS/UMC-iOS/Presentation/Login/JoinView/JoinCode.swift:
```diff
@@ -0,0 +1,85 @@
+                        Spacer().frame(height: 16)
+                        
+                        TextField("", text: $viewModel.inviteCode,
+                                  prompt: Text("운영진에게 받은 코드를 입력해주세요!")
```
<img width="372" alt="image" src="https://github.com/Team-UMC/UMC-iOS/assets/21362256/ed532602-3f7d-4e8e-aa62-1f0c4b4a0286">

이거 프롬프트가 너무 길어서 잘리니까 "초대코드를 입력해주세요!"로 글자 변경해주세요.

kyxxgsoo commented on UMC-iOS/UMC-iOS/Presentation/Login/Helper/DropDownPicker.swift:
```diff
@@ -0,0 +1,109 @@
+                                }
+                            }
+                    }
+                }
```
스크롤 바 조금 더 오른쪽에 옮기면 좋을 것 같습니다~

![image](https://github.com/Team-UMC/UMC-iOS/assets/21362256/f6d5c619-1bce-45d4-855d-f2847f4b6995)

이 부분 바로 밑에다 `.frame(minWidth: ???)`코드 적용시키면 해결 될거에요.

kyxxgsoo commented on UMC-iOS/UMC-iOS/Presentation/Login/Helper/JoinAgreCircleCheckButton.swift:
```diff
@@ -0,0 +1,31 @@
+    var body: some View {
+        Button(action: {
+            circleCheckButton_isChecked.toggle()
+            action()
```
이거 왜 넣은건가요??

leten02 replied:
이게 var action: () -> void를 넣어서 이 체크박스에 함수 내용을 사용 가능하게 만들어 외부 버튼에서 사용하게 하려고 만들었습니다.

예시) borderCustomButton(title: title, fontSize: 12)
> {
>                               viewModel.handleButtonSelection(title, selected: &viewModel.selectedGeneration)
>                             }
이 부분이 action이 사용되는 코드입니다!

kyxxgsoo commented on UMC-iOS/UMC-iOS/Presentation/Login/Helper/JoinCircleBorderButton.swift:
```diff
@@ -0,0 +1,38 @@
+        Button(action: {
+            // 버튼이 클릭되었을 때 수행할 동작
+            isButtonPressed.toggle()
+            action()
```
여기도 마찬가지로 action 왜 넣었는지 설명 가능할까여

kyxxgsoo commented on UMC-iOS/UMC-iOS/Presentation/Login/Helper/JoinNavigationButton.swift:
```diff
@@ -0,0 +1,38 @@
+
+    var body: some View {
+        Button(action: {
+            action()
```
이부분도여

leten02 replied:
이부분 action은 사용하지 않아 삭제하였습니다.

leten02 committed:
[Refactor]:codereview 수정
1.selectGeneration 중복코드 제거
2. joincode view 프롬프트 글자 변경
3. DropDownPicker 스크롤바 위치 수정
4. joinNavigationButton action 제거

yaeun reviewed (approved):
확인했습니다
